certbot-postfix/certbot_postfix/installer.py

```diff
@@ -0,0 +1,290 @@
+                                            self.conf('config-dir'))
+
+        # Ensure current configuration is valid.
+        self.config_test()
+
+        # Check Postfix version
+        self._check_version()
+        self._lock_config_dir()
+        self.install_ssl_dhparams()
+
+    def config_test(self):
+        """Test to see that the current Postfix configuration is valid.
+
+        :raises errors.MisconfigurationError: If the configuration is invalid.
+        """
+        self.postfix.test()
+
+    def _check_version(self):
+        """Verifies that the installed Postfix version is supported.
+
+        :raises errors.NotSupportedError: if the version is unsupported
+        """
+        if self._get_version() < constants.MINIMUM_VERSION:
+            version_string = '.'.join([str(n) for n in constants.MINIMUM_VERSION])
+            raise errors.NotSupportedError('Postfix version must be at least %s' % version_string)
+
+    def _lock_config_dir(self):
+        """Stop two Postfix plugins from modifying the config at once.
+
+        :raises .PluginError: if unable to acquire the lock
+        """
+        try:
+            certbot_util.lock_dir_until_exit(self.conf('config-dir'))
+        except (OSError, errors.LockError):
+            logger.debug("Encountered error:", exc_info=True)
+            raise errors.PluginError(
+                "Unable to lock %s" % self.conf('config-dir'))
+
+    def more_info(self):
+        """Human-readable string to help the user. Describes steps taken and any relevant
+        info to help the user decide which plugin to use.
+
+        :rtype: str
+        """
+        return (
+            "Configures Postfix to try to authenticate mail servers, use "
+            "installed certificates and disable weak ciphers and protocols.{0}"
+            "Server root: {root}{0}"
+            "Version: {version}".format(
+                os.linesep,
+                root=self.conf('config-dir'),
+                version='.'.join([str(i) for i in self._get_version()]))
+        )
+
+    def _get_version(self):
+        """Return the version of Postfix, as a tuple. (e.g. '2.11.3' is (2, 11, 3))
+
+        :returns: version
+        :rtype: tuple
+
+        :raises errors.PluginError: Unable to find Postfix version.
+        """
+        mail_version = self.postconf.get_default("mail_version")
+        return tuple(int(i) for i in mail_version.split('.'))
+
+    def get_all_names(self):
+        """Returns all names that may be authenticated.
+
+        :rtype: `set` of `str`
+
+        """
+        return certbot_util.get_filtered_names(self.postconf.get(var)
+                   for var in ('mydomain', 'myhostname', 'myorigin',))
+
+
+    def _set_vars(self, var_dict):
+        """Sets all parameters in var_dict to config file.
+        """
+        for param, acceptable in six.iteritems(var_dict):
+            if not util.is_acceptable_value(param, self.postconf.get(param), acceptable):
```

Comment:
Are you planning on always using tuples in `constants.py` for single values? I think that's fine, but if you are I think the body of this if statement can simply be `self.postconf.set(param, acceptable[0], acceptable)` because all the values are tuples now.